packages/gatsby/src/services/initialize.ts

```diff
@@ -407,34 +409,20 @@ export async function initialize({
       // }
       // }
 
-      if (
-        process.env.GATSBY_EXPERIMENTAL_PRESERVE_FILE_DOWNLOAD_CACHE ||
-        process.env.GATSBY_EXPERIMENTAL_PRESERVE_WEBPACK_CACHE
-      ) {
-        const deleteGlobs = [
-          // By default delete all files & subdirectories
-          `${cacheDirectory}/**`,
-          `${cacheDirectory}/*/`,
-        ]
-
-        if (process.env.GATSBY_EXPERIMENTAL_PRESERVE_FILE_DOWNLOAD_CACHE) {
-          // Stop the caches directory from being deleted, add all sub directories,
-          // but remove gatsby-source-filesystem
-          deleteGlobs.push(`!${cacheDirectory}/caches`)
-          deleteGlobs.push(`${cacheDirectory}/caches/*`)
-          deleteGlobs.push(`!${cacheDirectory}/caches/gatsby-source-filesystem`)
-        }
-
-        if (process.env.GATSBY_EXPERIMENTAL_PRESERVE_WEBPACK_CACHE) {
-          // Add webpack
-          deleteGlobs.push(`!${cacheDirectory}/webpack`)
-        }
-        await del(deleteGlobs)
-      } else {
-        // Attempt to empty dir if remove fails,
-        // like when directory is mount point
-        await fs.remove(cacheDirectory).catch(() => fs.emptyDir(cacheDirectory))
+      const deleteGlobs = [
+        // By default delete all files & subdirectories
+        `${cacheDirectory}/**`,
+        `!${cacheDirectory}/data`,
+        `${cacheDirectory}/data/**`,
+        `!${cacheDirectory}/data/gatsby-core-utils/`,
+        `!${cacheDirectory}/data/gatsby-core-utils/**`,
+      ]
+      if (process.env.GATSBY_EXPERIMENTAL_PRESERVE_WEBPACK_CACHE) {
```

Comment:
`GATSBY_EXPERIMENTAL_PRESERVE_WEBPACK_CACHE` is already set to `true` as a default, so I think you can skip this check and also add it to `deleteGlobs`

I assume you removed the file download cache because this + https://github.com/gatsbyjs/gatsby/pull/34758 adds it?

Reply:
I think we need to handle "old" location even if we do the move in the next PR to actually use mutex, as core has to work with potentialy order versions of `gatsby-source-filesystem`?